Standards/scs-03XX-v1-standard-roles.md

```diff
@@ -0,0 +1,203 @@
+
+The current RBAC rework in upstream OpenStack[^2] describes a "project-manager" persona utilizing a new "manager" role on project scope to perform more privileged operations than "member" (see "Phase 3" of the document).
+This role is intended to be used across various OpenStack services.
+As of the OpenStack release 2024.1 this role is not implemented in any of the core services yet, only in Ironic with `enforce_new_defaults` enabled[^3].
+
+On the other hand, the SCS project is making use of this role to implement a Domain Manager persona (see the [SCS Domain Manager standard under "Related Documents"](#scs-domain-manager-standard)).
+This persona will be available as a native upstream feature in Keystone starting with the 2024.2 release of OpenStack.
+
+As a result, the "manager" role has no effect outside of the Keystone Identity API until phase 3 of the RBAC rework is implemented upstream but can be used for identity-related functionality in Keystone.
+
+[^3]: [Implementation of the "manager" role in Ironic for the 2024.1 release](https://github.com/openstack/ironic/blob/stable/2024.1/ironic/common/policy.py#L76-L82)
+
+## Standard
+
+### Roles
+
+This standard establishes the following default roles in SCS clouds.
+The roles mentioned below MUST be present in the Identity API at all times.
+
+- reader
+- member
+- manager
+- admin
+- service
+
+#### Role Definitions
+
+The following overview will explain the roles' purposes and target scopes.
+Roles marked as "internal" are roles only meant to be assigned to technical user accounts intended for internal use by OpenStack services.
+
+Core Roles:
+
+| Role | Primary Target(s) | Purpose |
+|---|---|---|
+| reader | customer | read-only access to resources in the scope of authentication (e.g. project) |
+| member | customer | read and write access to resources in the scope of authentication (e.g. project) |
+| manager | customer | identity self-service capability within a domain, to assign/revoke roles between users, groups and projects |
+| admin | CSP | cloud-level global administrative access to all resources (cross-domain, cross-project) |
+| service | internal | internal technical user role for service communication |
+
+### API configuration
+
+All API services MUST be configured to use the Secure RBAC role model by enabling `enforce_new_defaults` and `enforce_scope` of the oslo.policy library.
+
+If custom policy rules to any API by a CSP, the following MUST be adhered to:
+
+1. The `policy_file` option of the oslo.policy library MUST be set to the name of the policy override file and not rely on default paths.
+2. The custom policy rules MUST NOT extend the privileges of the roles mentioned in this standard.
+
+Example configuration entries:
+
+```ini
+[oslo_policy]
+enforce_new_defaults = True
+enforce_scope = True
+policy_file = policy.yaml
+```
+
+#### API Policies
+
+TODO: what does the CSP need to adhere to when it comes to API policy configuration?
```

Comment:
Maybe this is more kind of an implementation note?
Because policy adjustments might occur more often than a change of the role model itself.

And the policy I am thinking about is the network rbac - which might be changed in the external network standard to only allow admin role to use it: https://github.com/SovereignCloudStack/standards/pull/572

Reply:
This is the last part I need to figure out for the standard: what to put here.

We will definitely need to make sure that CSPs apply the special policy templates offered by Octavia[^1] so that the LBaaS API will be aligned with the reader, member, admin model.

Other than that, we're pretty much in line with the in-code defaults now that we can use Secure RBAC. I need to double check but I think there won't be much more needed to be added here in terms of strict guidelines.

[^1]: https://docs.openstack.org/octavia/2024.1/configuration/policy.html#openstack-default-roles-policy-override-file

Reply:
I added Octavia there now plus some general do's and dont's. It's pretty brief and I don't think it justifies an implementation note split currently.

> And the policy I am thinking about is the network rbac - which might be changed in the external network standard to only allow admin role to use it: https://github.com/SovereignCloudStack/standards/pull/572

@josephineSei I'm wondering whether this should be part of the corresponding network standard or this role standard. I think it's more a detail of how the networking stuff works and is to be used, not a general role model or scoping thing. So I'm leaning more towards addressing this in the network standards if there is a network standard that would fit.